### PR TITLE
Add preference to enable vertical writing mode for form controls

### PR DESCRIPTION
**Pull request by nt1m**
#### 1a0c22eba39aa6bfeab16c9455210f5cb5e751e0
<pre>
Add preference to enable vertical writing mode for form controls
<a href="https://bugs.webkit.org/show_bug.cgi?id=248198">https://bugs.webkit.org/show_bug.cgi?id=248198</a>
rdar://102591869

Reviewed by Aditya Keerthi and Cameron McCormack.

Only append the rule forcing horizontal controls when the pref is disabled.

* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):
* Source/WTF/Scripts/Preferences/WebPreferencesExperimental.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/css/horizontalFormControls.css: Added.
(input, textarea, select, button, meter, progress):
* Source/WebCore/css/html.css:
(input, textarea, select, button, meter, progress): Deleted.
* Source/WebCore/style/InspectorCSSOMWrappers.cpp:
(WebCore::Style::InspectorCSSOMWrappers::collectDocumentWrappers):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebCore/style/UserAgentStyle.h:

Canonical link: <a href="https://commits.webkit.org/256928@main">https://commits.webkit.org/256928@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/7d110f7ce6957161f8ef4ba26737a777469caaef

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97266 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6530 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/30413 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/106784 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/167053 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/101234 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/6822 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35264 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89659 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/103468 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102929 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/5126 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/83892 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/32113 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/86977 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/88801 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/75042 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/88178 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/548 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20274 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/83627 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/531 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/21731 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/28467 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4776 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5331 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/44219 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/86336 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/1774 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41055 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/19449 "Passed tests") |

### CODE REVIEW

File: Source/WebCore/style/InspectorCSSOMWrappers.cpp [hunk 1]
```diff
@@ -122,7 +122,10 @@ void InspectorCSSOMWrappers::collectDocumentWrappers(ExtensionStyleSheets& exten
         collectFromStyleSheetContents(UserAgentStyle::svgStyleSheet);
         collectFromStyleSheetContents(UserAgentStyle::mathMLStyleSheet);
         collectFromStyleSheetContents(UserAgentStyle::mediaControlsStyleSheet);
+        collectFromStyleSheetContents(UserAgentStyle::horizontalFormControlsStyleSheet);
+#if ENABLE(FULLSCREEN_API)
```
Comment (heycam) [inline Source/WebCore/style/InspectorCSSOMWrappers.cpp]: Seems like the #if belongs in a different patch.

File: Source/WebCore/CMakeLists.txt [hunk 2]
```diff
@@ -1727,6 +1727,7 @@ set(WebCore_USER_AGENT_STYLE_SHEETS
     ${WEBCORE_DIR}/css/plugIns.css
     ${WEBCORE_DIR}/css/quirks.css
     ${WEBCORE_DIR}/css/svg.css
+    ${WEBCORE_DIR}/css/horizontalFormControls.css
```
Comment (heycam) [inline Source/WebCore/CMakeLists.txt]: Nit: keep this sorted.

And did you forget to add this file?

File: Source/WebCore/css/horizontalFormControls.css [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/* Form controls don't go vertical. */
```
Comment (heycam) [inline Source/WebCore/css/horizontalFormControls.css:27]: Does this need a big copyright comment block at the top? (Seems most UA sheet files do, but some don't.)

File: Source/WebCore/css/horizontalFormControls.css [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+input, textarea, select, button, meter, progress {
```
Comment (heycam) [inline Source/WebCore/css/horizontalFormControls.css:28]: I think you need this:

```
@namespace "http://www.w3.org/1999/xhtml";
```

**Review by heycam — APPROVED** [review]
r=me with those comments addressed.

**Review by pxlcoder — APPROVED** [review]
It might be good to A/B test this, since [the last time](https://trac.webkit.org/changeset/273839/webkit) I tried to add a new forms-affecting stylesheet by default, there was a 1% Speedometer regression.

Though this stylesheet is much simpler than the one I added.

File: Source/WebCore/CMakeLists.txt (hunk 2 above)
Comment (nt1m) [reply Source/WebCore/CMakeLists.txt]: Fixed.

File: Source/WebCore/css/horizontalFormControls.css (hunk 3 above)
Comment (nt1m) [reply Source/WebCore/css/horizontalFormControls.css:27]: Added (though this file is only meant to be temporary)

File: Source/WebCore/css/horizontalFormControls.css (hunk 4 above)
Comment (nt1m) [reply Source/WebCore/css/horizontalFormControls.css:28]: Added

File: Source/WebCore/style/InspectorCSSOMWrappers.cpp (hunk 1 above)
Comment (nt1m) [reply Source/WebCore/style/InspectorCSSOMWrappers.cpp]: Fixed.

**Comment by nt1m** [comment]
> It might be good to A/B test this, since [the last time](https://trac.webkit.org/changeset/273839/webkit) I tried to add a new forms-affecting stylesheet by default, there was a 1% Speedometer regression.

> Though this stylesheet is much simpler than the one I added.

Seems like the other changeset caused a regression due to ruleset duplication, this patch just moves a rule, so I suspect it should be fine.

I'll launch a A/B task once I am able to (right now the list of spades seems broken).
